lib/engine/game/g_18_esp/game.rb

```diff
@@ -472,8 +472,8 @@ def find_and_remove_train_for_minor(train_id, buyable = true)
 
         def init_company_abilities
           northern_corps = @corporations.select { |c| north_corp?(c) }
-          random_corporation = northern_corps[rand % northern_corps.size]
-          another_random_corporation = northern_corps[rand % northern_corps.size]
+          random_corporation = northern_corps[(rand / 100) % northern_corps.size]
+          another_random_corporation = northern_corps[(rand / 100) % northern_corps.size]
```

Comment:
This might well have a problem with differences in behaviour between the Ruby and Opal versions of the code. In Ruby `rand / 100` will be integer division and return an integer, but in Opal you will get a floating point number.

If you've got problems using the least significant bits of `rand` then use a bit shift operator, as suggested on Slack.

Reply:
i can't get bit shifting to work. I always end up with an even number. 

Would something like this work in Opal?
rand.to_s[0..-3].to_i

I just want to remove the last 3 digits of the rand number, this seems to make it rand enough for this applicaiton that i don't get a pattern

Reply:
nvm, i was shifting left and not right. Looks good. fixing it to >> 12

Reply:
That looks better.
I suggest you add a comment explaining why the bit shift is necessary. There's nothing in the code itself that makes this obvious.